File: app/src/main/java/com/tiktokaitool/ResultActivity.java
package com.tiktokaitool;
import android.content.Intent;
import android.graphics.*;
import android.os.Bundle;
import android.view.View;
import android.widget.*;
import androidx.appcompat.app.AppCompatActivity;

public class ResultActivity extends AppCompatActivity {
    private ImageView ivPreview;
    private TextView tvResult, tvStatus;
    private String imagePath, resultText;
    private PrefsManager prefs;
    private HistoryManager history;

    @Override protected void onCreate(Bundle b) {
        super.onCreate(b);
        setContentView(R.layout.activity_result);
        prefs   = new PrefsManager(this);
        history = new HistoryManager(this);
        imagePath = getIntent().getStringExtra("path");
        ivPreview = findViewById(R.id.iv_preview);
        tvResult  = findViewById(R.id.tv_result);
        tvStatus  = findViewById(R.id.tv_status);

        // Load preview with animation
        if (imagePath != null) {
            Bitmap bmp = BitmapFactory.decodeFile(imagePath);
            if (bmp != null) {
                ivPreview.setAlpha(0f);
                ivPreview.setImageBitmap(bmp);
                ivPreview.animate().alpha(1f).setDuration(300).start();
            }
        }

        // Check for cached result
        String cached = getIntent().getStringExtra("cached_result");
        if (cached != null && !cached.isEmpty()) {
            showResult(cached); return;
        }

        findViewById(R.id.btn_back).setOnClickListener(v -> finish());
        findViewById(R.id.btn_share).setOnClickListener(v -> share());
        findViewById(R.id.btn_retry).setOnClickListener(v -> doAnalyze());
        doAnalyze();
    }

    private void doAnalyze() {
        String key = prefs.getApiKey();
        if (key.isEmpty()) { tvResult.setText("No API key. Go to Settings → enter your Claude API key."); return; }
        tvStatus.setVisibility(View.VISIBLE);
        tvStatus.setText("Analyzing with Claude AI...");
        tvResult.setText("");
        tvResult.setAlpha(0f);
        findViewById(R.id.btn_share).setVisibility(View.GONE);
        findViewById(R.id.btn_retry).setVisibility(View.GONE);

        ClaudeApi.analyze(key, prefs.getModel(), prefs.getPrompt(), imagePath, new ClaudeApi.Callback() {
            @Override public void onSuccess(String r) {
                runOnUiThread(() -> { resultText=r; showResult(r); history.save(imagePath,r); });
            }
            @Override public void onError(String e) {
                runOnUiThread(() -> {
                    tvStatus.setVisibility(View.GONE);
                    tvResult.setText("Error: "+e+"\n\nTap Retry to try again.");
                    tvResult.animate().alpha(1f).setDuration(300).start();
                    findViewById(R.id.btn_retry).setVisibility(View.VISIBLE);
                });
            }
        });
    }

    private void showResult(String text) {
        tvStatus.setVisibility(View.GONE);
        tvResult.setText(text);
        tvResult.animate().alpha(1f).setDuration(400).start();
        findViewById(R.id.btn_share).setVisibility(View.VISIBLE);
        findViewById(R.id.btn_retry).setVisibility(View.VISIBLE);
    }

    private void share() {
        if (resultText == null) return;
        Intent i = new Intent(Intent.ACTION_SEND);
        i.setType("text/plain"); i.putExtra(Intent.EXTRA_TEXT, resultText);
        startActivity(Intent.createChooser(i, "Share analysis"));
    }
}

File: app/src/main/java/com/tiktokaitool/ui/ResultActivity.kt
package com.tiktokaitool.ui

import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import android.content.Intent
import android.graphics.BitmapFactory
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.tiktokaitool.databinding.ActivityResultBinding
import io.noties.markwon.Markwon

class ResultActivity : AppCompatActivity() {

    private lateinit var binding: ActivityResultBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityResultBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val result = intent.getStringExtra("analysis_result") ?: "No result"
        val imgPath = intent.getStringExtra("cropped_image_path")

        // Show cropped image
        imgPath?.let {
            try {
                val bmp = BitmapFactory.decodeFile(it)
                binding.ivCroppedPreview.setImageBitmap(bmp)
            } catch (e: Exception) {
                // ignore
            }
        }

        // Render markdown result
        try {
            val markwon = Markwon.create(this)
            markwon.setMarkdown(binding.tvResult, result)
        } catch (e: Exception) {
            binding.tvResult.text = result
        }

        binding.toolbar.setNavigationOnClickListener { finish() }

        binding.btnCopyResult.setOnClickListener {
            val cm = getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
            cm.setPrimaryClip(ClipData.newPlainText("AI Result", result))
            Toast.makeText(this, "✅ Copied to clipboard!", Toast.LENGTH_SHORT).show()
        }

        binding.btnShareResult.setOnClickListener {
            val shareIntent = Intent(Intent.ACTION_SEND).apply {
                type = "text/plain"
                putExtra(Intent.EXTRA_TEXT, "🤖 TikAI Finder Result:\n\n$result")
            }
            startActivity(Intent.createChooser(shareIntent, "Share Result"))
        }

        binding.btnFindAgain.setOnClickListener {
            startActivity(Intent(this, CropAnalyzeActivity::class.java))
            finish()
        }

        binding.btnSaveHistory.setOnClickListener {
            // Save to history via PrefsManager
            com.tiktokaitool.utils.HistoryManager(this).saveEntry(result, imgPath)
            Toast.makeText(this, "✅ Saved to history!", Toast.LENGTH_SHORT).show()
            binding.btnSaveHistory.isEnabled = false
            binding.btnSaveHistory.text = "✅ Saved"
        }
    }
}
